### PR TITLE
[WIP] Ngb pagination

### PR DESCRIPTION
**Pull request by VictorADS**
Hello guys, I'd like to get your help on this one.
Currently the pagination works and we can switch between them.
The problem is that in the URL is encoded and  the comma ','  for the parameter 'sort'  is changed.
Thus the parseAscending method  in PaginationUtil doesn't work.

Should I change PaginationUtil method ? 
Or should I try to have a raw url (ie : not encoded) ?

### CODE REVIEW

**Comment by deepu105** [comment]
@geraldhumphries I think you did the sorting stuff rite? do you have an opinion here?
@VictorADS if its easy to modify the util method to accept encoded URL then go for it

File: generators/client/templates/angular/src/main/webapp/app/shared/_shared.module.ts [hunk 1]
```diff
@@ -58,7 +57,7 @@ import {
         <%_ if (enableSocialSignIn) { _%>
         SocialService,
         <%_ } _%>
-        AuthService
+        AuthService,
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/_shared.module.ts]: stray comma here

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/_user-management.component.html [hunk 2]
```diff
@@ -78,9 +78,9 @@ <h2 jhi-translate="userManagement.home.title">Users</h2>
         </table>
     </div>
     <%_ if (databaseType !== 'cassandra') { _%>
-    <div class="text-xs-center">
+    <div class="text-xs-center" *ngIf="users">
         <jhi-item-count [page]="page" [total]="queryCount" [items-per-page]="itemsPerPage"></jhi-item-count>
-        <ngb-pagination size="sm" [collectionSize]="totalItems" [(page)]="page" [pageSize]="itemsPerPage" (click)="loadPage(page)"></ngb-pagination>
```
Comment (geraldhumphries) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/_user-management.component.html]: `size="sm"` is removed here, which is a change to the UI. Unless this is an obsolete/unused attribute I would leave it in.

File: generators/client/templates/angular/src/main/webapp/app/shared/service/_pagination-util.service.ts [hunk 3]
```diff
@@ -8,7 +8,10 @@ export class PaginationUtil  {
         if (sortArray.length > 1){
             return sort.split(',').slice(-1)[0] === 'asc';
         } else {
-            // default to true if no sort defined
```
Comment (geraldhumphries) [inline generators/client/templates/angular/src/main/webapp/app/shared/service/_pagination-util.service.ts]: I would readd this comment above `return true;`

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/_user-management.component.html (hunk 2 above)
Comment (VictorADS) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/_user-management.component.html]: Now it is set via the PaginationConfig file

File: generators/client/templates/angular/src/main/webapp/app/shared/service/_pagination-util.service.ts [hunk 4]
```diff
@@ -8,7 +8,10 @@ export class PaginationUtil  {
...
+            sortArray = sort.split('%2C');
+            if (sortArray.length > 1){
+                return sort.split('%2C').slice(-1)[0] === 'asc';
+            }
```
Comment (geraldhumphries) [inline generators/client/templates/angular/src/main/webapp/app/shared/service/_pagination-util.service.ts]: This code is essentially a duplicate of the rest of the code in this function. It should be refactored to be cleaner. Also, return `sortArray.slice(-1)[0] === 'asc';` unless there's a reason for it to split the string twice (I know that line was preexisting).

**Comment by geraldhumphries** [comment]
I never considered encoded URLs so if this fixes a bug involving that then I'm all for it.

**Pull request merged**